### ISSUE TITLE
0.10.0-rc1 : Segment Loading

### ISSUE THREAD

**Issue by naman-aasaan** [druid-io/druid#4025]
I recently started using this latest version and there are a few things happening very strange with respect to segment loading.

Any exception for any of the task and it starts getting this exception continuously and it stops any of the further segments coming for the same data source. I can see this issue happening on all my environments where i have upgraded with this version.
```
2017-03-08T16:58:53,264 WARN [ZkCoordinator-0] com.metamx.common.RetryUtils - Failed on try 1, retrying in 946ms.
java.io.FileNotFoundException: File does not exist: /druid/segments/IL2S/20170308T100000.000Z_20170308T110000.000Z/2017-03-08T16_00_01.006Z/0_
index.zip
        at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:71)
        at org.apache.hadoop.hdfs.server.namenode.INodeFile.valueOf(INodeFile.java:61)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocationsInt(FSNamesystem.java:1828)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1799)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1712)
        at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:588)
        at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtocolServerSide
```

**Comment by gianm** [comment]
@naman-aasaan, is this happening on historical nodes? Is it on rc1 or have you built your own version including the fix in #4003?

**Comment by gianm** [comment]
Also does that stack trace keep going? If so could you include the whole thing?

**Comment by akashdw** [comment]
@gianm @naman-aasaan We ran into a very similar situation before fix in #4003. In short that bug caused bad entries in metadata.
1. This bug caused mm’s to think other instance of realtime wrote data into hdfs and tasks ended up writing segment metadata with hdfs location of non existing paths. 
2. Coordinator assigns those bad entries to Historical.
3. Historical's were not able to pull data from those (non existing)paths and coordinator keeps assigning those segments to some other historical.

This behavior continued even after the druid 0.10.0 revert i.e coordinator assigns those segments (b/c of bad entries) to some historical's and ended up spending most of the time on those assignments. we updated the metadata entries in mysql and did a backfill to recover from this corruption.
